vmdb/app/presenters/menu/default_menu.rb

```diff
@@ -34,28 +34,38 @@ def clouds_menu_section
-          Menu::Item.new('ems_cluster',      N_('Clusters'),         'ems_cluster',   {:feature => 'ems_cluster_show_list'},   '/ems_cluster'),
-          Menu::Item.new('host',             N_('Hosts'),            'host',          {:feature => 'host_show_list'},          '/host'),
+          Menu::Item.new('ems_cluster',      clusters_name,          'ems_cluster',   {:feature => 'ems_cluster_show_list'},   '/ems_cluster'),
+          Menu::Item.new('host',             hosts_name,             'host',          {:feature => 'host_show_list'},          '/host'),
           Menu::Item.new('vm_infra',         N_('Virtual Machines'), 'vm_infra_explorer_accords',
-                                                                                     {:feature => 'vm_infra_explorer_accords', :any => true},
-                                                                                                                              '/vm_infra/explorer'),
+                                                                                      {:feature => 'vm_infra_explorer_accords', :any => true},
+                                                                                                                               '/vm_infra/explorer'),
           Menu::Item.new('resource_pool',    N_('Resource Pools'),   'resource_pool', {:feature => 'resource_pool_show_list'}, '/resource_pool'),
           Menu::Item.new('storage',          ui_lookup(:tables => 'storages'),
-                                                                    'storage',       {:feature => 'storage_show_list'},       '/storage'),
+                                                                     'storage',       {:feature => 'storage_show_list'},       '/storage'),
           Menu::Item.new('repository',       N_('Repositories'),     'repository',    {:feature => 'repository_show_list'},    '/repository'),
           Menu::Item.new('pxe',              N_('PXE'),              'pxe',           {:feature => 'pxe', :any => true},       '/pxe/explorer'),
           Menu::Item.new('miq_request_host', N_('Requests'),         nil,             {:feature => 'miq_request_show_list'},   '/miq_request?typ=host'),
-          Menu::Item.new('provider_foreman',
-                         N_('Configuration Management'),
-                         'provider_foreman_explorer_accords',
-                         {:feature => 'provider_foreman_explorer_accords',
-                          :any     => true},
-                         '/provider_foreman/explorer')
+          Menu::Item.new('provider_foreman', N_('Configuration Management'), 'provider_foreman_explorer_accords',
+                         {:feature => 'provider_foreman_explorer_accords', :any => true}, '/provider_foreman/explorer')
         ])
       end
 
+      def hybrid_name(klass, name1, name2, name3)
+        lambda do
+          case klass.node_types
```

Comment:
@tenderlove : not known until run-time ; actually until the menu item is shown; therefor the lambda

I agree with @Fryguy about not loving the #2708 changes. I feel we should not need to do the SQL queries fired at each menu redraw.